test/utils/asserts.hpp

```diff
@@ -23,10 +24,8 @@ MATCHER_P(Result, value, "")  // NOLINT(readability-redundant-string-init)
     if (arg.trapped || !arg.has_value)
         return false;
 
-    if constexpr (std::is_same_v<value_type, float>)
-        return arg.value.f32 == value;
-    else if constexpr (std::is_same_v<value_type, double>)
-        return arg.value.f64 == value;
+    if constexpr (std::is_floating_point_v<value_type>)
+        return arg.value.template as<value_type>() == fizzy::test::FP{value};
```

Comment:
This C++ syntax of `template` just to be able to use `as<value_type>` is terribly, but it is what it is.

Reply:
Why is this beneficial over the previous one, btw?

Reply:
The `FP` is comparing binary representation.

The `template` is a mandatory hint to the compiler that the `<` in `as<value_type>` is a template bracket. Fortunately, GCC is giving nice hints when you forget it.